### PR TITLE
authentication changes

### PR DESCRIPTION
**Pull request by PythonDeveloperVipul**

### CODE REVIEW

File: authentication/apps.py [hunk 1]
```diff
@@ -4,3 +4,5 @@
 class AuthConfig(AppConfig):
     default_auto_field = 'django.db.models.BigAutoField'
     name = 'authentication'
+    def ready(self):
```
Comment (priteshtagline) [inline authentication/apps.py:7]: What is the use of this?

File: authentication/serializers.py [hunk 2]
```diff
@@ -31,13 +31,8 @@ def validate(self, attrs):
...
 
     def create(self, validated_data):
-        user = User.objects.create(
-            first_name=validated_data['first_name'],
-            last_name=validated_data['last_name'],
-            email=validated_data['email'],
-        )
-
+        validated_data.pop('confirm_password')
+        user = super(RegisterSerializer, self).create(validated_data)
```
Comment (priteshtagline) [inline authentication/serializers.py:37]: Shouldn't we have to validate the serializer first and then create it?

File: authentication/utils.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from django.core.mail import send_mail
+
+
+def send_mail_reset_password(default_from_email, email_plaintext_message, reset_password_token):
```
Comment (priteshtagline) [inline authentication/utils.py]: Function parameters should be subject, message, optional from email, to_email.

File: backend/settings.py [hunk 4]
```diff
@@ -130,12 +135,15 @@
...
 
-# Email config credential
-EMAIL_HOST = 'smtp.gmail.com'
-EMAIL_PORT = 587
-EMAIL_HOST_USER = 'vipul.tagline@gmail.com'
-EMAIL_HOST_PASSWORD = 'tagline@123'
-EMAIL_USE_TLS = True
+# Email config
+
+EMAIL_BACKEND = os.getenv('EMAIL_BACKEND')
```
Comment (priteshtagline) [inline backend/settings.py]: This should be static and there is no need to read it from .env

File: requirements.txt [hunk 5]
```diff
@@ -1 +1,5 @@
-Django==4.0.4
\ No newline at end of file
+Django==4.0.4
+djangorestframework
+djangorestframework-simplejwt
```
Comment (priteshtagline) [inline requirements.txt]: How you have frozen the requirements? It is not containing the version of the package?

File: authentication/apps.py (hunk 1 above)
Comment (PythonDeveloperVipul) [reply authentication/apps.py:7]: signal used

File: authentication/serializers.py (hunk 2 above)
Comment (PythonDeveloperVipul) [reply authentication/serializers.py:37]: yes, validate method already check validation

**Pull request closed without merging**